src/datasources/cache/cache.first.data.source.ts

```diff
@@ -45,16 +47,42 @@ export class CacheFirstDataSource {
     const cached = await this.cacheService.get(cacheDir);
     if (cached != null) {
       winston.debug(`[Cache] Cache hit: [${cacheDir.key}, ${cacheDir.field}]`);
-      return JSON.parse(cached);
+      const data = JSON.parse(cached);
+      if (this.isNotFoundError(data)) {
+        throw new NotFoundException(data.message);
+      }
+      return data;
+    }
+    try {
+      winston.debug(`[Cache] Cache miss: [${cacheDir.key}, ${cacheDir.field}]`);
+      const { data } = await this.networkService.get(url, params);
+      const rawJson = JSON.stringify(data);
+      await this.cacheService.set(cacheDir, rawJson, expireTimeSeconds);
+      return data;
+    } catch (error) {
+      if (this.isNotFoundError(error)) {
+        await this.cacheError(cacheDir, error);
+      }
+      throw error;
     }
-    winston.debug(`[Cache] Cache miss: [${cacheDir.key}, ${cacheDir.field}]`);
-    const { data } = await this.networkService.get(url, params);
-    const rawJson = JSON.stringify(data);
-    await this.cacheService.set(
-      new CacheDir(cacheDir.key, cacheDir.field),
-      rawJson,
-      expireTimeSeconds,
-    );
-    return data;
+  }
+
+  private isNotFoundError(data: unknown) {
+    const error = data as NetworkResponseError;
```

Comment:
Out of curiosity, what is your opinion on trying to get the expected field without the cast?

Since `NetworkResponseError` is not the only error that can be thrown maybe we should try to get the expected field without specifying the class itself?